Ical.Net/Serialization/DataTypes/RecurrencePatternSerializer.cs

```diff
@@ -244,7 +244,20 @@ public override object Deserialize(TextReader tr)
                 var keywordPairs = match.Groups[2].Value.Split(';');
                 foreach (var keywordPair in keywordPairs)
                 {
+                    if (keywordPair.Length == 0)
+                    {
+                        // This is illegal but ignore for now.
+                        continue;
+                    }
+
                     var keyValues = keywordPair.Split('=');
+                    if (keyValues.Length != 2)
+                    {
+                        // ArgumentExceptions seem to be the Exception of choise for this class. Should
+                        // probably be changed to a more specific exception type.
+                        throw new ArgumentException($"The recurrence rule contains an invalid part.");
```

Comment:
As we know it here, show the input value causing the exception (and maybe what's wrong with it)

Reply:
I was thinking that too but then intentionally left out the actual input data, since the data could contain sensitive information (in the end a calendar can be something very personal). I know that in other places we include the data, so if you think we should, I'll be happy to add it.

Reply:
For developers the generic message would be less helpful than the detailed one. So my vote is for the more helpful way :)

Reply:
The old conflict ;-)

Reply:
added some details